sql/rpl_rli_pdb.cc

```diff
@@ -1437,16 +1437,12 @@ ulong circular_buffer_queue::en_queue(void *item)
 
 void* circular_buffer_queue::head_queue()
 {
-  uchar *ret= NULL;
   if (entry == size)
   {
     DBUG_ASSERT(len == 0);
+    return NULL;
```

Comment:
Yes, the implementation of head_queue was not working - nobody has checked that as this function was never used. get_dynamic copies element of an array into argument void *element passed to it. However, it requires that this element has allocated memory. In previous implementation we were always passing NULL to it - thus sig fault. We do not need to copy the element though, we only need access to it - so I have changed it to dynamic_array_ptr - which just returns the pointer. If a caller of head_queue needs a copy of this memory he can copy it on his own.